src/tests/audio/tasks/data_augmentation_test.py

```diff
@@ -0,0 +1,39 @@
+"""Module for testing data augmentation on audios."""
+
+import torch
+from torch_audiomentations import Compose, PolarityInversion
+
+from senselab.audio.tasks.data_augmentation import augment_audio_dataset
+from senselab.utils.data_structures.audio import Audio
+from senselab.utils.data_structures.datasets import SenselabDataset
+
+
+def test_audio_data_augmentation() -> None:
+    """Test data augmentations using the new Audio data types."""
+    apply_augmentation = Compose(transforms=[PolarityInversion(p=1, output_type="dict")], output_type="dict")
+
+    audio_paths = [
+        "src/tests/data_for_testing/audio_48khz_mono_16bits.wav",
+        "src/tests/data_for_testing/audio_48khz_stereo_16bits.wav",
+    ]
+    audio_dataset_from_paths = SenselabDataset(audios=audio_paths)
+    mono_audio, stereo_audio = audio_dataset_from_paths.create_audio_split_for_pydra_task()
+    mono_inverted = augment_audio_dataset(mono_audio, apply_augmentation)
+    stereo_inverted = augment_audio_dataset(stereo_audio, apply_augmentation)
+    assert torch.equal(
+        mono_audio[0].waveform, -1 * mono_inverted[0].waveform
+    ), "Audio should have been inverted by the augmentation"
+    assert torch.equal(
+        stereo_audio[0].waveform, -1 * stereo_inverted[0].waveform
+    ), "Audio should have been inverted by the augmentation and not affected by stereo audio"
+
+    batched_audio = SenselabDataset(
+        audios=[
+            Audio(waveform=stereo_audio[0].waveform[0], sampling_rate=stereo_audio[0].sampling_rate),
+            Audio(waveform=stereo_audio[0].waveform[1], sampling_rate=stereo_audio[0].sampling_rate),
+        ]
+    ).create_audio_split_for_pydra_task(2)
+    batch_inverted = augment_audio_dataset(batched_audio[0], apply_augmentation)
```

Comment:
have you tested this code with and without GPU?

Reply:
no, as I mentioned before, I am not sure exactly how we would go about realistically testing on GPU. We could mock it, but that seems like it doesn't really test